changelog.d/12829.bugfix

```diff
@@ -0,0 +1,2 @@
+Fixed a bug where device list updates over federation would fail if they
+were not a list. Contributed by Carl Bordum Hansen.
```

Comment:
```suggestion
Fix a bug where we did not correctly handle invalid device list updates over federation. Contributed by Carl Bordum Hansen.
```

This gets shoved verbatim into the changelog, so we try and keep them as oneliners